src/lib/comms/include/sol-oic-client.h

```diff
@@ -328,59 +332,52 @@ bool sol_oic_client_get_server_info_by_addr(struct sol_oic_client *client,
  *        macro. @a data is the user's @a callback_data. When timeout is reached
  *        and no packet has arrived, callback is called with @c NULL @a addr and
  *        @c NULL repr_vec so any clean up can be performed.
- *
  * @param callback_data User's data to be passed to @a callback.
  *
- * @return True if packet was successfully sent. False otherwise.
+ * @return @c 0 on success or a negative number on errors.
  */
-bool sol_oic_client_resource_request(struct sol_oic_client *client, struct sol_oic_resource *res,
-    sol_coap_method_t method,
-    bool (*fill_repr_map)(void *data, struct sol_oic_map_writer *repr_map),
-    void *fill_repr_map_data,
-    void (*callback)(sol_coap_responsecode_t response_code, struct sol_oic_client *cli, const struct sol_network_link_addr *addr,
-    const struct sol_oic_map_reader *repr_vec, void *data),
-    const void *callback_data);
+int sol_oic_client_resource_request(struct sol_oic_client *client, struct sol_oic_client_request *request, void (*callback)(sol_coap_responsecode_t response_code, struct sol_oic_client *cli, const struct sol_network_link_addr *addr, const struct sol_oic_map_reader *repr_vec, void *data), const void *callback_data);
```

Comment:
I didn't change that in this patch and I think this should be done in other commit. I updated the issue to fix OIC API (https://github.com/solettaproject/soletta/issues/1721) adding this.